playbook-base.yaml

```diff
@@ -1,76 +1,160 @@
+        state: present
+        name: [
+          # Container runtime
+          cri-o, crictl, crun,
+          # Kubelet & kubectl
+          kubelet, kubectl,
+          # Kubeadm is needed in worker nodes as it provides /usr/lib/systemd/system/kubelet.service.d/10-kubeadm.conf
+          kubeadm,
+          # NFS
+          nfs-utils,
+          # Misc node administration tools.
+          btop, htop, lsof, nano
+        ]
 
-  - name: Copy folder structure
-    ansible.builtin.file:
-      path: "/{{ item.path }}"
-      state: directory
-      recurse: yes
-    with_community.general.filetree: "{{ playbook_dir }}/files/"
-    loop_control:
-      label: "/{{ item.path }}"
-    when: item.state == 'directory'
+    - name: Copy folder structure
+      ansible.builtin.file:
+        path: "/{{ item.path }}"
+        state: directory
+        recurse: yes
+      with_community.general.filetree: "{{ playbook_dir }}/files/"
+      loop_control:
+        label: "/{{ item.path }}"
+      when: item.state == 'directory'
+    - name: Copy and render template files
+      ansible.builtin.template:
+        src: "{{ item.src }}"
+        dest: "/{{ item.path | regex_replace('\\.jinja2$', '') }}"
+      with_community.general.filetree: "{{ playbook_dir }}/files/"
+      loop_control:
+        label: "/{{ item.path | regex_replace('\\.jinja2$', '') }}"
+      when: item.state == 'file' and item.path is regex('\\.jinja2$')
+    - name: Copy static files
+      ansible.builtin.copy:
+        src: "{{ item.src }}"
+        dest: "/{{ item.path }}"
+      with_community.general.filetree: "{{ playbook_dir }}/files/"
+      loop_control:
+        label: "/{{ item.path }}"
+      when: item.state == 'file' and item.path is not regex('\\.jinja2$')
 
-  - name: Copy and render template files
-    ansible.builtin.template:
-      src: "{{ item.src }}"
-      dest: "/{{ item.path | regex_replace('\\.jinja2$', '') }}"
-    with_community.general.filetree: "{{ playbook_dir }}/files/"
-    loop_control:
-      label: "/{{ item.path | regex_replace('\\.jinja2$', '') }}"
-    when: item.state == 'file' and item.path is regex('\\.jinja2$')
+    # Kubelets are configured to _not_ use systemd-resolved.
+    - name: Systemd-resolved is disabled
+      ansible.builtin.systemd:
+        name: systemd-resolved
+        state: stopped
+        enabled: false
+        masked: true
 
-  - name: Copy files
-    ansible.builtin.copy:
-      src: "{{ item.src }}"
-      dest: "/{{ item.path }}"
-    with_community.general.filetree: "{{ playbook_dir }}/files/"
-    loop_control:
-      label: "/{{ item.path }}"
-    when: item.state == 'file' and item.path is not regex('\\.jinja2$')
+    - name: CRI-O is started and enabled
+      ansible.builtin.systemd:
+        name: crio
+        state: started
+        enabled: yes
+    - name: Kubelet is enabled  # Storage nodes override kubelet.service's WantedBy.
+      ansible.builtin.systemd:
+        name: crio
+        enabled: yes
```

Comment:
Why not started?

Reply:
kbecause hete it should say `kubelet` instead of crio`. Good catch!